grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -992,12 +1013,10 @@ class ProductController {
      */
     def addSynonymToProduct = {
         println "addSynonymToProduct() " + params
-        Product product = null
+        Product product = Product.read(params.id)
```

Comment:
it can't stay like this - as discussed, it will cause `LazyInitializationExceptions` when trying to add a wrong synonym (when the validation should appear)

Reply:
Thanks for pointing that out, this didn't look like a problem to me at first but I see it now